### DISCUSSION TITLE
Add ColorResource and allow to pass in instead of Color

### DISCUSSION THREAD

**Discussion by me2beats** [godotengine/godot-proposals#5354] [Editor]
The idea is to have something like "color references". You just create a ColorResource, set it's rgba and then pass it to Inspector, for example in some theme overrides.
And when you want to replace the color in all places you set — you simply change the resource color.

**Comment by me2beats** [comment]
Not sure if it should be editor only feature, you may want to set such color references from code as well.
But in this case idk how to do this without breaking typing I mean the property type will be "Color or ColorResource" which looks like a Union type

**Comment by dalexeev** [comment]
> The idea is to have something like "color references". You just create a ColorResource, set it's rgba and then pass it to Inspector, for example in some theme overrides. And when you want to replace the color in all places you set — you simply change the resource color.

Use `Theme`.

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot-proposals/issues/1734.
